lib/debug.ml

```diff
@@ -185,16 +185,19 @@ let logs_reporter =
   in
   let report src level ~over k msgf =
     let formatter ?header ?tags fmt =
+      let buf = Buffer.create 80 in
+      let buf_fmt = Format.formatter_of_buffer buf in
       let k _ =
-        let msg = Format.flush_str_formatter () in
+        Format.pp_print_flush buf_fmt ();
+        let msg = Buffer.contents buf in
         (* We map the Logs source name to the "brand", so we have to use the
            name of the Logs source when enabling/disabling it *)
         let brand = Logs.Src.name src in
         output_log brand (logs_to_syslog_level level) (logs_level_to_priority level) msg;
         over ();
         k ()
```

Comment:
Just a note here, that this calls the outer `k` because `k` is not `let rec`, so no infinite loop here.